src/SafeCurl.php

```diff
@@ -56,6 +56,18 @@ public function execute($url) {
         do {
             $url = $this->urlValidator->validateUrl($url);
 
+            if ($url && isset($url['host']) && isset($url['ips'])) {
+                $resolves = [];
+                foreach ($url['ips'] as $url_ip) {
+                    foreach ([80, 443, 8080] as $url_port) {
```

Comment:
I don't think we need to put all the standard ports in here. We should be able to determine the port from the URL. We're only pinning the DNS resolution for this request, so this is looping for the sake of pinning at least two more than we need.

Reply:
It's possible for the URL to use a non-standard port. We default to these standard three, but the library could be used in such a way that additional ports could be whitelisted. As it stands, we're only pinning the three standards, so no pinning would occur for a URL with a non-standard port.